File: 程序员宝典/JavaScript实现/02_02_返回倒数第 k 个节点.js
// 1、快慢指针
var kthToLast = function(head, k) {
    let slow = head, fast = head;

    for (let i = 0; i < k; ++i) {
        fast = fast.next;
    }

    while (fast) {
        slow = slow.next;
        fast = fast.next;
    }

    return slow.val;
};

File: 程序员宝典/C++实现/02.02. 返回倒数第 k 个节点.cpp
// 1、快慢指针
class Solution {
public:
    int kthToLast(ListNode* head, int k) {
        ListNode *slow = head, *fast = head;

        for (int i = 0; i < k; ++i) {
            fast = fast->next;
        }

        while (fast) {
            slow = slow->next;
            fast = fast->next;
        }

        return slow->val;
    }
};